src/main/java/com/rocketden/main/service/GameManagementService.java

```diff
@@ -147,6 +147,22 @@ public void setStartGameTimer(Game game, Long duration) {
         gameTimer.getTimer().schedule(endGameTimerTask, duration * 1000);
     }
 
+    // Test the submission, return the results, and send a socket update
+    public SubmissionDto runCode(String roomId, SubmissionRequest request) {
+        Game game = getGameFromRoomId(roomId);
+
+        if (request.getInitiator() == null || request.getCode() == null || request.getLanguage() == null || request.getInput() == null) {
```

Comment:
Looks good - a small nit I'd have would just be to add one more API test for this endpoint

Reply:
Sounds good, I'll add this now. 👍